geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/AccumuloIndexAdapter.scala

```diff
@@ -0,0 +1,137 @@
+/***********************************************************************
+* Copyright (c) 2013-2016 Commonwealth Computer Research, Inc.
+* All rights reserved. This program and the accompanying materials
+* are made available under the terms of the Apache License, Version 2.0
+* which accompanies this distribution and is available at
+* http://www.opensource.org/licenses/apache2.0.php.
+*************************************************************************/
+
+package org.locationtech.geomesa.accumulo.index
+
+import java.util.Map.Entry
+
+import org.apache.accumulo.core.client.IteratorSetting
+import org.apache.accumulo.core.data.{Key, Mutation, Range, Value}
+import org.apache.hadoop.io.Text
+import org.geotools.factory.Hints
+import org.locationtech.geomesa.accumulo.data.{AccumuloDataStore, AccumuloFeature}
+import org.locationtech.geomesa.accumulo.index.AccumuloIndexAdapter.ScanConfig
+import org.locationtech.geomesa.accumulo.iterators._
+import org.locationtech.geomesa.index.api.{FilterStrategy, QueryPlan}
+import org.locationtech.geomesa.index.index.IndexAdapter
+import org.locationtech.geomesa.utils.collection.CloseableIterator
+import org.locationtech.geomesa.utils.index.VisibilityLevel
+import org.opengis.feature.simple.{SimpleFeature, SimpleFeatureType}
+import org.opengis.filter.Filter
+
+trait AccumuloIndexAdapter extends IndexAdapter[AccumuloDataStore, AccumuloFeature, Mutation, Range] {
+
+  this: AccumuloFeatureIndex =>
+
+  import org.locationtech.geomesa.utils.geotools.RichSimpleFeatureType.RichSimpleFeatureType
+
+  override protected def createInsert(row: Array[Byte], feature: AccumuloFeature): Mutation = {
+    val mutation = new Mutation(row)
+    feature.fullValues.foreach(v => mutation.put(v.cf, v.cq, v.vis, v.value))
+    if (hasPrecomputedBins) {
+      feature.binValues.foreach(v => mutation.put(v.cf, v.cq, v.vis, v.value))
+    }
+    mutation
+  }
+
+  override protected def createDelete(row: Array[Byte], feature: AccumuloFeature): Mutation = {
+    val mutation = new Mutation(row)
+    feature.fullValues.foreach(v => mutation.putDelete(v.cf, v.cq, v.vis))
+    if (hasPrecomputedBins) {
```

Comment:
yeah, i thought it would probably be a wash with the overhead of executing a functional val or whatever.